docs/adr/0006-recording-activity-log-data-within-the-account.md

```diff
@@ -0,0 +1,174 @@
+
+We would take whichever occurs first during a session (identified by a unique `session_id` produced by auth), and use that to create a "Signed in event". The timestamp will also be taken of whichever event triggers creation. Where the trigger is an `AUTH_AUTH_CODE_ISSUED` we will also be able to add an initial "visited" service.
+
+After this point every subsequent `AUTH_AUTH_CODE_ISSUED` event will be used to signal that a user has "Visited" a service, enhancing their log of events.
+
+## Architecture of the system
+
+The system consists of a mix of lambdas, SQS and dynamoDB for persistence.
+
+### Query Activity Log
+
+Will take a new event from TxMA from [the raw event store](./0004-store-all-event-data.md) and query the activity log store for a record with a matching `user_id` and `session_id`.
+
+This differs from the User Service store in it's scoping to session. We are interested in updating records here where there is continued activity from within a session, otherwise it is just a "new activity log event".
+
+The query lambda retrieves data then passes both the TxMA event and any retrieved data on to an SQS queue.
+
+It will also filter out events that are not deemed relevant to producing signed in activity logs, specifically we will only take events from the Raw Event Store dynamoDB stream that are of type:
+
+- `AUTH_AUTH_CODE_ISSUED`
+- `AUTH_IPV_AUTHORISATION_REQUESTED`
+
+Any other events will be discarded.
+
+### Format User Services
+
+Will take the TxMA event and any retrieved data and either:
+
+- Create a new event if nothing was retrieved with that `session_id`. In this case this is the first event we have detected for that session, and so would be treated as a new "Signed In Event" irrespective of if they are `AUTH_AUTH_CODE_ISSUED` or `AUTH_IPV_AUTHORISATION_REQUESTED`. This will create a new "Signed In" Log Event
+- If data has been returned by `session_id` then after that point any authentication (`AUTH_AUTH_CODE_ISSUED` event) will be treated as the user having "visited" a service, resulting in an update to the queried event with data from TxMA.
+
+After the data structures have been reconciled it will be passed on to the write lambda.
+
+### Write User Services
+
+Will take a single formatted service record and attempt to write it to DynamoDB.
+
+### SQS Queues
+
+Each lambda will be separated by an SQS Queue.
+Each lambda will have it's own Dead Letter Queue.
+
+Data on services will be stored against a User ID that we can match to the Account Management RP's session. Data will be persisted in the following format:
+
+### Service record data structure
+
+```typescript
+type UrnFdnSub = string;
+type ClientId = string;
+
+export interface UserData {
+  user_id: UrnFdnSub;
+  govuk_signin_journey_id: string;
```

Comment:
Should we include the session_id here - https://github.com/alphagov/di-account-management-backend/pull/95/files#diff-3d97f46fb56381153aaf510597651dc011f9911b6bb87e3f380207992e986ec8R8